### PR TITLE
feat(tile-select)!: Updates input display prop nomenclature

### PR DESCRIPTION
**Pull request by macandcheese**
**Related Issue:** #1445 

## Summary
Contains breaking changes.

The “showInput” prop has been broken apart into “inputEnabled” (boolean), and “inputAlignment”. “Left” / “Right” has been renamed to start / end to better match other components nomenclature.

This means that input-enabled will now be FALSE by default, so users will have to opt-in to display the input if desired. The alignment of the input when displayed still defaults to start (previously was “left”).

Additionally, the component now uses `getElementDir` to add dir tag to host, and RTL styling has been fixed.

Updates tests

Updates stories

### CODE REVIEW

**Review by driskull — APPROVED** [review]
🏆 👍

File: src/components/calcite-tile-select/calcite-tile-select.stories.ts [hunk 1]
```diff
@@ -24,8 +24,8 @@ export const Light = (): string => html`
     heading="${text("heading", "Tile heading lorem ipsum")}"
     ${boolean("hidden", false)}
     icon="${select("icon", iconNames, iconNames[296])}"
-    name="${text("name", "tile-select-demo")}"
-    show-input="${select("show-input", ["left", "right", "none"], "left")}"
+    ${boolean("show-input", false)}
+    show-input="${select("input-alignment", ["start", "end"], "start")}"
```
Comment (jcfranco) [inline src/components/calcite-tile-select/calcite-tile-select.stories.ts:28]: Attr needs to be updated, this is setting the alignment on `show-input`. There are other instances of this.

File: src/components/calcite-tile-select/calcite-tile-select.e2e.ts [hunk 2]
```diff
@@ -25,10 +25,10 @@ describe("calcite-tile-select", () => {
       { propertyName: "heading", value: "My Tile Select Heading" },
       { propertyName: "hidden", value: true },
       { propertyName: "icon", value: "layers" },
+      { propertyName: "inputAlignment", value: "end" },
```
Comment (jcfranco) [inline src/components/calcite-tile-select/calcite-tile-select.e2e.ts:28]: Nitpick: one value should suffice for this test. Just needs a valid value to do the check.

**Review by jcfranco — APPROVED** [review]
Nice! 

I noticed that we use `hide/show` and `disable/enable` in some props. Should we stick to one vs the other?

Also, can you hold off on installing this until next week?

File: src/components/calcite-tile-select/calcite-tile-select.stories.ts (hunk 1 above)
Comment (macandcheese) [reply src/components/calcite-tile-select/calcite-tile-select.stories.ts:28]: Whoops, good catch.

File: src/components/calcite-tile-select/calcite-tile-select.e2e.ts (hunk 2 above)
Comment (macandcheese) [reply src/components/calcite-tile-select/calcite-tile-select.e2e.ts:28]: Will remove this and other duplicate values in these tests.

**Comment by macandcheese** [comment]
Yep - can hold off on installing. Re: show vs. enable... That's valid, this uses the existing name but since it's a breaking change anyway we could reconsider. `enable-input`? Looks like "show-*" is used in one other place to show the back button in panel, so that could get updated as well to enable.

**Review by eriklharper — APPROVED** [review]

**Comment by macandcheese** [comment]
@jcfranco cool to update this prop name to “enableInput” and try and get this in? 

Test failure seems unrelated to this PR.

**Comment by jcfranco** [comment]
> @jcfranco cool to update this prop name to “enableInput” and try and get this in?

👍

**Comment by macandcheese** [comment]
Screener failures were related to `enableInput` not being true by default in Storybooks. Verified locally that you can still enable those in Storybook.

**Comment by macandcheese** [comment]
Hold on merging... sounds like we may be renaming this prop to `input-enabled` as part of a larger nomenclature change.

**Comment by macandcheese** [comment]
Is `input-enable` the right name here, post discussion @jcfranco ?

**Comment by jcfranco** [comment]
👍

**Review by eriklharper — APPROVED** [review]

**Review by jcfranco — APPROVED** [review]
Aside from @eriklharper's comment, this LGTM!

File: src/components/calcite-tile-select-group/calcite-tile-select-group.stories.ts [hunk 3]
```diff
@@ -95,7 +100,7 @@ export const Dark = (): string => html`
     heading="Tile title lorem ipsum Tile title lorem ipsum Tile title lorem ipsum Tile title lorem ipsum Tile title lorem ipsum Tile title lorem ipsum"
     icon="layers"
     name="dark"
-    show-input="${select("show-input", ["left", "right", "none"], "left")}"
+    input-alignment="${select("input-enabled", ["left", "right", "none"], "left")}"
```
[the remark this answers is not in the document]
Comment (macandcheese) [reply src/components/calcite-tile-select-group/calcite-tile-select-group.stories.ts:103]: Ah, yes. The prop names have changed too many times in this PR, ha 😦 . Will fix.

**Comment by julio8a** [comment]
What's the status of this PR? Can we fix the conflicts and merge it in?

**Comment by paulcpederson** [comment]
build failure is unrelated, I'm going to merge this in.

**Pull request merged**